skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,148 @@
+
+    Subtract a white background from an example image
+
+    >>> image = data.page()
+    >>> result, bg = rolling_ball(image, radius=100, white_background=True)
+
+    Notes
+    -----
+    Check out the gallery example of this function (TODO: link)
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+    """
+
+    if not isinstance(radius, float):
+        try:
+            radius = float(radius)
+        except ValueError:
+            raise ValueError(f"Radius should be float, was {type(radius)}")
+
+    if radius <= 0:
+        raise ValueError(f"Radius must be greater zeros, was {radius}")
+
+    try:
+        white_background = bool(white_background)
+    except ValueError:
+        raise ValueError(
+            f"white_background must be convertible to a boolean, "
+            f"was {type(white_background)}"
+        )
+
+    if not isinstance(input_img, np.ndarray):
+        raise ValueError(
+            f"input_img must be a np.ndarray, "
+            f"was {type(input_img)}"
+        )
+
+    if not issubclass(input_img.dtype.type, np.integer):
+        raise ValueError("Currently only integer images are supported.")
+
+    img = input_img.copy()
+    if white_background:
+        img = invert(img)
+
+    # sagitta assuming the position is where the ball touches
+    # the image umbra
+    L = np.arange(-radius, radius + 1)
+    X, Y = np.meshgrid(L, L)
+    distance = np.sqrt(X ** 2 + Y ** 2)
+    sagitta = radius - np.sqrt(radius ** 2 - distance ** 2)
+    sagitta[np.isnan(sagitta)] = np.Inf
+
+    kernel = np.array((X ** 2 + Y ** 2) <= radius ** 2, dtype=float)
+    kernel[kernel == 0] = np.Inf
+
+    kernel_size = np.array(kernel.shape)
+    img_original = img.copy()
+    img_size = img.shape
+    x = img
+    stride = (1, 1)
+
+    # pad the image
+    padding_amount = kernel_size // 2 * 2
+    half_pad = padding_amount // 2
+    img = np.Inf * np.ones(x.shape + padding_amount, dtype=x.dtype)
+    img[half_pad[0]:-half_pad[0], half_pad[1]:-half_pad[1]] = x
+    large_img_size = img.shape
+
+    # indices of kernel in image coords
+    x_idx = np.arange(kernel_size[1])
+    y_idx = large_img_size[1] * np.arange(kernel_size[0])
+    kernel_idx = (x_idx[np.newaxis, :] + y_idx[:, np.newaxis]).flatten()
+
+    # indices corresponding to each ancor of the kernel
+    # (top left corner instead of center)
+    x_ancors = np.arange(
+        large_img_size[1] - kernel_size[1] + 1, step=stride[1])
+    y_ancors = large_img_size[1] * \
+        np.arange(large_img_size[0] - kernel_size[0] + 1, step=stride[0])
+    ancor_offsets = (x_ancors[np.newaxis, :] +
+                     y_ancors[:, np.newaxis]).flatten()
+
+    # large images or kernel sizes don't fit into memory
+    # do it in batches instead
+    background = np.zeros(img_original.size)
+    batch_size = int(2 ** 10)
```

Comment:
This seems like it should be a parameter...

Reply:
likely. Still debating how to best add it to the function so that it makes sense; hence it being a magic number for now.

I am currently torn between exposing `batch_size` (easy to maintain, but might not make intuitive sense), or `max_memory` (easier to understand, but more code in the function). Thoughts?

Reply:
I'd prefer batch size.

Reply:
See my [comment](https://github.com/scikit-image/scikit-image/pull/4851#discussion_r462755574) above: if we change the implementation a bit, maybe we can use `chunk_size` rather than batch size.

Reply:
there now is a parameter called `chunk_size` which specifies the dimensions of image chunks that will be processed in batches.